src/cuisine.API/Repositories/ProductRepository.cs

```
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using cuisine.Entity;
+
+namespace cuisine.API
+{
+	//This class will interact with the data persistance.
+	public class ProductRepository : IService<Product>
+	{
+		//This instance will act as data persistance.
+		private List<Product> _productCollection = new List<Product>();
+
+
+		public void Add(Product product)
+		{
+			if(!string.IsNullOrEmpty(product.Name) && product.UnitaryPrice > 0)
```

Nit: You can throw an exception here is you want. You can do later if you need it.